katdal/sensordata.py

```diff
@@ -652,10 +652,12 @@ def __str__(self):
             names = sorted([key for key in self.keys()])
             maxlen = max([len(name) for name in names])
             objects = [self.get(name, extract=False) for name in names]
-        obj_reprs = [(("<numpy.ndarray shape=%s type=%s at 0x%x>" % (obj.shape, obj.dtype, id(obj)))
-                     if isinstance(obj, np.ndarray) else repr(obj)) for obj in objects]
-        actual = ['%s : %s' % (str(name).ljust(maxlen), obj_repr) for name, obj_repr in zip(names, obj_reprs)]
-        virtual = ['%s : <function %s.%s>' % (str(pat).ljust(maxlen), func.__module__, func.__name__)
+        obj_reprs = [(f"<numpy.ndarray shape={obj.shape} type={obj.dtype} at 0x{id(obj):x}>"
+                      if isinstance(obj, np.ndarray) else repr(obj)) for obj in objects]
+        actual = ['{} : {}'.format(str(name).ljust(maxlen), obj_repr)
```

Comment:
```suggestion
        actual = [f'{name!s:{maxlen}} : {obj_repr}'
```
The `!s` forces to string before applying the field width, which is important since it seems `:n` will left-just strings but right-justify numbers. On the other hand, I'm not sure why `name` wouldn't be a string already.

Reply:
I guess I wanted to make sure you could call `ljust` on `name`, but I agree that it should be a string.